packages/contracts-bedrock/test/L2/L1BlockCGT.t.sol

```diff
@@ -0,0 +1,156 @@
+contract L1BlockCGT_TestInit is CommonTest {
+    address depositor;
+    IL1BlockCGT l1BlockCGT;
+
+    /// @notice Sets up the test suite.
+    function setUp() public virtual override {
+        super.enableCustomGasToken();
+        super.setUp();
+        l1BlockCGT = IL1BlockCGT(address(l1Block));
+        depositor = l1BlockCGT.DEPOSITOR_ACCOUNT();
+    }
+}
+
+/// @title L1BlockCGT_Version_Test
+/// @notice Test contract for L1BlockCGT `version` function.
+contract L1BlockCGT_Version_Test is L1BlockCGT_TestInit {
+    /// @notice Tests that the version function returns a valid string. We avoid testing the
+    ///         specific value of the string as it changes frequently.
+    function test_version_succeeds() external view {
+        assert(bytes(l1BlockCGT.version()).length > 0);
+    }
+}
+
+/// @title L1BlockCGT_GasPayingToken_Test
+/// @notice Tests the `gasPayingToken` function of the `L1BlockCGT` contract.
+contract L1BlockCGT_GasPayingToken_Test is L1BlockCGT_TestInit {
+    /// @notice Tests that the `gasPayingToken` function reverts.
+    function test_gasPayingToken_deprecated_reverts() external {
+        vm.expectRevert("L1BlockCGT: deprecated");
+        l1BlockCGT.gasPayingToken();
+    }
+}
+
+/// @title L1BlockCGT_GasPayingTokenName_Test
+/// @notice Tests the `gasPayingTokenName` function of the `L1Block` contract with custom gas
+///         token enabled.
+contract L1BlockCGT_GasPayingTokenName_Test is L1BlockCGT_TestInit {
+    /// @notice Tests that the `gasPayingTokenName` function returns the correct token name.
+    function test_gasPayingTokenName_succeeds() external view {
+        assertEq(liquidityController.gasPayingTokenName(), l1BlockCGT.gasPayingTokenName());
+    }
+}
+
+/// @title L1BlockCGT_GasPayingTokenSymbol_Test
+/// @notice Tests the `gasPayingTokenSymbol` function of the `L1Block` contract with custom gas
+///         token enabled.
+contract L1BlockCGT_GasPayingTokenSymbol_Test is L1BlockCGT_TestInit {
+    /// @notice Tests that the `gasPayingTokenSymbol` function returns the correct token symbol.
+    function test_gasPayingTokenSymbol_succeeds() external view {
+        assertEq(liquidityController.gasPayingTokenSymbol(), l1BlockCGT.gasPayingTokenSymbol());
+    }
+}
+
+/// @title L1BlockCGT_IsCustomGasToken_Test
+/// @notice Tests the `isCustomGasToken` function of the `L1Block` contract with custom gas token
+///         enabled.
+contract L1BlockCGT_IsCustomGasToken_Test is L1BlockCGT_TestInit {
+    /// @notice Tests that the `isCustomGasToken` function returns false when no custom gas token
+    ///         is used.
+    function test_isCustomGasToken_succeeds() external view {
+        assertTrue(l1BlockCGT.isCustomGasToken());
+    }
+}
+
+/// @title L1BlockCGT_SetL1BlockValues_Test
+/// @notice Tests the `setL1BlockValues` function of the `L1Block` contract with custom gas token
+///         enabled.
+contract L1BlockCGT_SetL1BlockValues_Test is L1Block_SetL1BlockValues_Test {
```

Comment:
Since we're using the matrix option in CI, no longer necessary to have these inherited versions of the tests